src/main/java/cloud/orbit/actors/cluster/impl/lettuce/LettuceClient.java

```diff
@@ -0,0 +1,148 @@
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY ELECTRONIC ARTS AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ELECTRONIC ARTS OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cloud.orbit.actors.cluster.impl.lettuce;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.lettuce.core.KeyScanCursor;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.ScanArgs;
+import io.lettuce.core.SetArgs;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.codec.RedisCodec;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class LettuceClient<K, V>
+{
+    private static Logger logger = LoggerFactory.getLogger(LettuceClient.class);
+
+    private final RedisClient redisClient;
+    private final RedisCodec<K, V> codec;
+    private final String redisUri;
+
+    private RedisAsyncCommands<K, V> asyncCommands;
+
+    public LettuceClient(final String resolvedUri, RedisCodec<K, V> codec)
+    {
+        this.redisUri = resolvedUri;
+        this.redisClient = RedisClient.create(resolvedUri);
+        this.codec = codec;
+        this.asyncCommands = this.redisClient.connect(codec).async();
+    }
+
+    public String getRedisUri()
+    {
+        return this.redisUri;
+    }
+
+    public CompletableFuture<V> get(final K key) {
+        return this.asyncCommands.get(key).toCompletableFuture();
+    }
+
+    public CompletableFuture<String> set(final K key, final V value) {
+        return this.asyncCommands.set(key, value).toCompletableFuture();
+    }
+
+    public CompletableFuture<String> set(final K key, final V value, final long expireMs) {
+        if (expireMs < 1) {
+            return this.set(key, value);
+        }
+        return this.asyncCommands.set(key, value, SetArgs.Builder.px(expireMs)).toCompletableFuture();
+    }
+
+    public CompletableFuture<Long> del( final K key) {
+        return this.asyncCommands.del(key).toCompletableFuture();
+    }
+
+    public CompletableFuture<List<String>> scan(final String matches) {
+        // Batches of 1000?  TODO measure and adjust batch size if necessary
```

Comment:
How are you planning on testing this batch size? Should the batch size value be configurable instead of being hardcoded?

Reply:
The batch size is arbitrary, and needs to satisfy the directory size of the # of nodes we would run (~200) so 1000 is more than enough to get in a single call... if we ran more than 1000 nodes this would end up taking 2 calls to complete... ideally, this could change to a lazy paginated future pattern, but for now this is plenty.

Reply:
If you think the 1000 batch size is appropriate, then I would suggest removing the `TODO` comment.

If you think we will need to adjust the batch size moving forward, then I would suggest making it a configurable value so we don't need to release a new version of this project to update it.